packages/sanity/src/form/studio/inputResolver/helpers.test.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint max-nested-callbacks: */
+
+import {getObjectFieldLevel, type PartialObjectFieldProps} from './helpers'
```

Comment:
Rewriting to use createSchema, as it is not modelling the types correctly and the fix doesnt actually work 😆